### PR TITLE
Wingman: Destruct on empty case

### LINKED ISSUE #1716: case split on case of statement with no destructor cases declared
This isn't exactly Wingman, but I think @isovector is probably the man for the job (?) given what he's been working on...

```
case my expr of
```

In this case, we'll get some errors about allow "empty-case" and add "EmptyCase" etc. 

My suggestion: to add a code action that inserts empty case destructors:

```
case myMaybeValue of
```

becomes

```
case myMaybeValue of
   Nothing -> _
   Just val -> _
```

and now the same Wingman workflows are recovered.

### PR DESCRIPTION
**Pull request by isovector**
This PR adds a code lens to empty `case x of` expressions, offering to expand out their pattern matches. It works by looking for empty case matches, typechecking their scrutinee, and then grafting newly generated `Match` es back into the AST. Seems to work pretty well.

destruct on empty case

Fixes #1716

### CODE REVIEW

**Comment by wz1000** [comment]
I wonder if this would be better to implement as a completion rather than a code action?

File: plugins/hls-tactics-plugin/src/Wingman/Types.hs [hunk 1]
```diff
@@ -117,6 +117,9 @@ instance Show Class where
 instance Show (HsExpr GhcPs) where
   show  = unsafeRender
 
+instance Show (HsExpr GhcTc) where
+  show  = unsafeRender
+
 instance Show (HsDecl GhcPs) where
   show  = unsafeRender
```
Comment (pepeiborra) [inline plugins/hls-tactics-plugin/src/Wingman/Types.hs:122]: Please consider moving these to the Orphans module

**Review by pepeiborra — APPROVED** [review]
This is awesome! I have wanted this feature so many times!

File: plugins/hls-tactics-plugin/src/Wingman/Types.hs (hunk 1 above)
Comment (isovector) [reply plugins/hls-tactics-plugin/src/Wingman/Types.hs:122]: Happy to --- followup PR ok?

**Pull request merged**